libunsf.c

```diff
@@ -1041,14 +1020,37 @@ static char *unsf_concat(const char *s1, const char *s2) {
     char *result = NULL;
     if (!(result = (char *) malloc(len1 + len2 + 1))) { /* +1 for the zero-terminator */
         fprintf(stderr, "Memory allocation failed with mem size %lu\n", (long unsigned int) (len1 + len2 + 1));
-        exit(1);
+        exit(1); /* FIXME: library must NOT exit() */
     }
     memcpy(result, s1, len1);
     memcpy(result + len1, s2, len2 + 1);/* +1 to copy the null-terminator */
     return result;
 }
 
-static int unsf_mkdir(char *dir, mode_t mode) {
+#ifdef _WIN32
+static int sys_mkdir(const char *p) {
+    if (_mkdir(p) != -1) return 0;
+    if (errno == EEXIST) return 0;
+    return -1;
+}
+#elif defined(__SOME_FOO_PLATFORM__)
+static int sys_mkdir(const char *p) {
+#error implement me..
+    return -1;
+}
```

Comment:
This is a catch all for handling other platforms? If someone wants to add a platform, they extend it here?